src/execution/mod.rs

```diff
@@ -209,18 +211,59 @@ impl<'b> RuntimeInstance<'b> {
                 }
                 // i32.mul: [i32 i32] -> [i32]
                 0x6C => {
-                  let v1: i32 = stack.pop_value(ValType::NumType(NumType::I32)).into();
-                  let v2: i32 = stack.pop_value(ValType::NumType(NumType::I32)).into();
-                  let res = v1.wrapping_mul(v2);
+                    let v1: i32 = stack.pop_value(ValType::NumType(NumType::I32)).into();
+                    let v2: i32 = stack.pop_value(ValType::NumType(NumType::I32)).into();
+                    let res = v1.wrapping_mul(v2);
+
+                    trace!("Instruction: i32.mul [{v1} {v2}] -> [{res}]");
+                    stack.push_value(res.into());
+                }
+                // i32.div_s: [i32 i32] -> [i32]
+                0x6D => {
+                    let dividend: i32 = stack.pop_value(ValType::NumType(NumType::I32)).into();
+                    let divisor: i32 = stack.pop_value(ValType::NumType(NumType::I32)).into();
+
+                    if dividend == 0 {
+                        return Err(crate::Error::RuntimeError(
+                            crate::core::error::RuntimeError::DivideBy0,
+                        ));
```

Comment:
```rust
return Err(crate::Error::RuntimeError(
    crate::core::error::RuntimeError::DivideBy0,
));
```

This syntax is a bit too verbose. I would import the two error enums directly. resultin in
```rust
return Err(RuntimeError(DivideBy0));
```

Reply:
AFAIK in the diff this is still not resolved.

Reply:
Now it should be fixed